### ISSUE TITLE
CanvasVirtualControl OnRegionsInvalidated not triggered when launching app

### ISSUE THREAD

**Issue by tmjorud** [Microsoft/Win2D#479]
Hello :)

I'm building a UWP app which contains a CanvasVirtualControl on the MainPage. The control is drawing thin vertical lines horizontally across the screen with some space between each line, so the graphics is very simple.

Now, my problem is this. When I launch the app, the OnRegionsInvalidated event never fires, and the lines are not drawn on the canvas. The only way I can make the event fire, is when I resize the window, or I navigate to another page and then back to the MainPage.

I'm not sure how to troubleshoot this issue. Any help is most appreciated :) 

Here is my XAML markup for the control
```
<Grid>
        <ScrollViewer HorizontalScrollMode="Auto"
                      HorizontalScrollBarVisibility="Auto"
                      VerticalScrollBarVisibility="Disabled">
            <Grid>
                <canvas:CanvasVirtualControl x:Name="BarMarks"
                                             RegionsInvalidated="BarMarks_OnRegionsInvalidated"
                                             Width="10000" />
            </Grid>
        </ScrollViewer>
</Grid>
```

And here is my code for the OnRegionsInvalidated event. 
```
private void BarMarks_OnRegionsInvalidated(CanvasVirtualControl sender, CanvasRegionsInvalidatedEventArgs args)
       {
            foreach (var region in args.InvalidatedRegions)
            {
                using (var ds = sender.CreateDrawingSession(region))
                {
                    var left = ((int) (region.Left/100))*100;
                    var right = ((int) ((region.Right)/100))*100;
                    var height = BarMarks.ActualHeight;

                    for (var x = left; x <= right; x += 100)
                    {
                        var pos1 = new Vector2(x, (float) height);
                        var pos2 = new Vector2(x, 0);
                        ds.DrawLine(pos1, pos2, Colors.Aqua);
                    }
                }
            }
        }
```

**Comment by shawnhar** [comment]
Please share instructions on how to repro this issue  (ideally an app that we can run to see the problem).

**Comment by tmjorud** [comment]
Hi shawnhar!

Thank you very much for the quick respons. I've made an example uwp app that repro the issue, you can download the VS 2015 solution [here](https://1drv.ms/u/s!Avkf9elOeIJGg4pM9NUFS2FGW0MN7A)

When you run the app, you will see just a blank window. If you resize the windows height (**not width**) or if you maximise/restore the window, you will see the red vertical lines be drawn across the screen. It seems like as soon as the hight of the window is changed, the OnRegionsInvalidated is fired.

**Comment by shawnhar** [comment]
Thanks for the excellent repro!   I see the same problem.

**Comment by tmjorud** [comment]
Great! :) It seems like this issue is isolated to the CanvasVirtualControl only, as far as I know. I've not had this problem with either the CanvasControl or the CanvasAnimatedControl.

**Comment by mscherotter** [comment]
I just ran into this problem when creating a new Window that has a CanvasVirtualControl in it as well.  Code sample is here: https://1drv.ms/f/s!AmvW0E0DYMdm2Z1J1uOxZPSDWVLKWQ

**Comment by softwarenerd** [comment]
I have also just run into this issue in my Win2D project.

The CreateResources and RegionsInvalidated events are never fired. Invalidating the control using Invalidate() or Invalidate(Rect) does nothing, either.

Win2D is amazing, but this bug has stopped my development efforts cold. Hopefully there will be a speedy resolution.

**Comment by coder89** [comment]
I created a pull request that we confirmed solving this issue in our projects: #493

**Comment by VoteForPedro** [comment]
temp fix:   set clear color after load 

private void DrawSurface_Loaded(object sender, RoutedEventArgs e)
        {
            Dispatcher.RunIdleAsync((a) =>
            {
                canvas.ClearColor = Color.FromArgb(255, 255, 255, 255);
            });
        }

**Comment by shawnhar** [comment]
This is fixed in Win2D 1.21.0 (released today)
